src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/Filter';
+import { getTodos } from './api';
+
+const filteredTodos = (todos: Todo[], query: string, filterBy: FilterBy) => {
+  let preparedTodos = [...todos];
+
+  if (query) {
+    const preparedQuery = query.trim().toLowerCase();
+
+    preparedTodos = preparedTodos.filter(todo => todo.title.toLowerCase().includes(preparedQuery));
+  }
+
+  switch (filterBy) {
+    case FilterBy.ACTIVE:
+      preparedTodos = preparedTodos.filter(todo => !todo.completed);
+      break;
+    case FilterBy.COMPLETED:
+      preparedTodos = preparedTodos.filter(todo => todo.completed);
+      break;
+    case FilterBy.ALL:
+    default:
+      break;
+  }
+
+  return preparedTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+  const [isDataLoading, setIsDataLoading] = useState(true);
+
+  const visibleTodos = filteredTodos(todos, query, filterBy);
+
+  const fetchData = async () => {
+    try {
+      const data = await getTodos();
+
+      setTodos(data);
+      setIsDataLoading(false);
+      setIsLoadingError(false);
+    } catch (error) {
+      setIsLoadingError(true);
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const onShowTodo = useCallback((todo: Todo) => {
+    setSelectedTodo(todo);
```

Comment:
why do you need to select todo?
Just make fetch use when you click to the button